desidlas/datasets/preprocess.py

```diff
@@ -0,0 +1,325 @@
+    #open the csv file
+    with open(output_dir, mode=mode,newline="") as summary_table:
+        summary_table_writer = csv.DictWriter(summary_table,headers)
+        if mode == "w":
+            summary_table_writer.writeheader()
+        for sightline in sightlines:
+            #for each sightline, read its information and write to the csv file
+            info = {"id":sightline.id, "z_qso":sightline.z_qso, "s2n": sightline.s2n,"wavelength_start_b":10**sightline.loglam[0],
+                    "wavelength_end_b":10**sightline.loglam[sightline.split_point_br-1],"pixel_start_b":0,"pixel_end_b":sightline.split_point_br-1,
+                    "wavelength_start_r":10**sightline.loglam[sightline.split_point_br],"wavelength_end_r":10**sightline.loglam[sightline.split_point_rz-1],
+                "pixel_start_r":sightline.split_point_br,"pixel_end_r":sightline.split_point_rz-1,"wavelength_start_z":10**sightline.loglam[sightline.split_point_rz],
+                    "wavelength_end_z":10**sightline.loglam[-1],"pixel_start_z":sightline.split_point_rz,"pixel_end_z":len(sightline.loglam)-1}
+            
+            dlas_col_density = ""
+            dlas_central_wavelength = ""
+            for dla in sightline.dlas:
+                dlas_col_density += str(dla.col_density)+","
+                dlas_central_wavelength += str(dla.central_wavelength)+","
+            info["dlas_col_density"] = dlas_col_density[:-1]
+            info["dlas_central_wavelength"] = dlas_central_wavelength[:-1]
+            
+            #write to the csv file
+            summary_table_writer.writerow(info)
+#from dla_cnn.desi.DesiMock import DesiMock
+def write_summary_table(nums, version,path, output_path):
+    """
+    Directly read data from fits files and write the summary table, the summary table contains all available sightlines(dlas!=[] and z_qso>2.33) in the given fits files.
+    -----------------------------------------------------------------------------------------------------------------------------------------
+    parameters:
+    nums: list, the given fits files' id, its elements' format is int, and one should make sure all fits files are available before invoking this funciton, otherwise some sightlines can be missed;
+    version: int, the version of the data set we use, e.g. if the version is v9.16, then version = 16
+    path: str, the dir of the folder which stores the given fits file, the folder's structure is like folder-fits files' id - fits files , if you are still confused, you can check the below code about read data from the fits file;
+    output_path: str, the dir where the summary table is generated, and if there have been a summary table, then we will remove it and generate a new summary table;
+    ------------------------------------------------------------------------------------------------------------------------------------------
+    retrun:
```

Comment:
typo

Reply:
This function has no returns, so should I just delete the return

Reply:
yes, please do

Reply:
Done